sfge/src/main/java/com/salesforce/graph/symbols/SharingPolicySymbolProviderVertexVisitor.java

```diff
@@ -0,0 +1,73 @@
+package com.salesforce.graph.symbols;
+
+import com.salesforce.exception.ProgrammingException;
+import com.salesforce.graph.symbols.apex.ApexValue;
+import com.salesforce.graph.vertex.*;
+import com.salesforce.rules.ops.boundary.SharingPolicyBoundary;
+import com.salesforce.rules.ops.boundary.SharingPolicyBoundaryDetector;
+import java.util.Optional;
+import org.apache.tinkerpop.gremlin.process.traversal.dsl.graph.GraphTraversalSource;
+
+public class SharingPolicySymbolProviderVertexVisitor extends DefaultSymbolProviderVertexVisitor {
+
+    private final SharingPolicyBoundaryDetector boundaryDetector;
+
+    public SharingPolicySymbolProviderVertexVisitor(
+            GraphTraversalSource g,
+            SharingPolicyBoundaryDetector boundaryDetector,
```

Comment:
The `SharingPolicyBoundaryDetector` (boundary manager?) has to be shared between this class and the `UseWithSharingOnDatabaseOperationVisitor` class. This class pushes/pulls the sharing policy boundaries to the stack, and that class checks the current boundary. Are there any better solutions than just giving both classes the same `SharingPolicyBoundaryDetector` in the constructor?

Reply:
We looked at this and determined that since `DefaultSymbolProviderVertexVisitor` has its own logic for visiting nodes, best not to mess with it in a child.